op-batcher/batcher/driver.go

```diff
@@ -469,12 +471,33 @@ func (l *BatchSubmitter) sendTransaction(ctx context.Context, txdata txData, que
 		candidate = l.calldataTxCandidate(data)
 	}
 
-	intrinsicGas, err := core.IntrinsicGas(candidate.TxData, nil, false, true, true, false)
-	if err != nil {
-		// we log instead of return an error here because txmgr can do its own gas estimation
-		l.Log.Error("Failed to calculate intrinsic gas", "err", err)
-	} else {
-		candidate.GasLimit = intrinsicGas
+	if *candidate.To != l.RollupConfig.BatchInboxAddress {
+		return fmt.Errorf("candidate.To is not inbox")
+	}
+	if l.inboxIsEOA == nil {
+		var l2Client dial.EthClientInterface
+		l2Client, err = l.EndpointProvider.EthClient(ctx)
+		if err != nil {
+			return fmt.Errorf("EthClient failed:%w", err)
+		}
+		var code []byte
+		code, err = l2Client.CodeAt(ctx, *candidate.To, nil)
+		if err != nil {
+			return fmt.Errorf("CodeAt failed:%w", err)
+		}
+		isEOA := len(code) == 0
+		l.inboxIsEOA = &isEOA
+	}
+
+	// only set GasLimit when inbox is EOA so that later on `EstimateGas` will be called
+	if *l.inboxIsEOA {
```

Comment:
When `inboxIsEOA`, what is the GasLimit of the batchTx?

Reply:
`core.IntrinsicGas(candidate.TxData, nil, false, true, true, false)`, the same as the GasLimit before this change.

Reply:
Sorry. I mean `!inboxInEOA` case.

Reply:
It will be set later [here](https://github.com/ethereum-optimism/optimism/blob/8f516faf42da416c02355f9981add3137a3db190/op-service/txmgr/txmgr.go#L264-L278) by `EstimateGas`.